types/react/test/tsx.tsx

```diff
@@ -721,3 +721,49 @@ function elementTypeTests() {
     <RenderReactNode />;
     React.createElement(RenderReactNode);
 }
+
+function managingRefs() {
+    const genericRefBad = React.useRef<Element>();
+    // $ExpectType Element | undefined
+    genericRefBad.current;
```

Comment:
I don't actually know what people expect this type to be. Sometimes it sounds like it should just be `Element | null`, or maybe `Element | null | undefined` and `<div ref />` should just allow this type. 

Asked in https://twitter.com/sebsilbermann/status/1639925689997565956 so maybe we can settle this. Because to me this is what you want at declaration-time. But given the usage, you actually want `useRef<Element>(null)`. And it seems impossible to reconcile where the mistake is for all possible usages of `useRef`: at usage or at declaration. The best "fix" for this would be having custom TS error messages that explain that you probably wanted `useRef<Element>(null)` since the ref is only managed by React.

Unless we want to be prescriptive in React that only `null` is a bottom-value and disallow `useRef()` or `useRef(undefined)` alltogether.